### ISSUE TITLE
Optimize `verify_logs` by using `get_block_receipts`

### ISSUE THREAD

**Issue by eshaan7** [a16z/helios#438]
Can significantly optimize `verify_logs` too. Will create a separate issue for it once this PR is merged.

_Originally posted by @eshaan7 in https://github.com/a16z/helios/pull/423#discussion_r1836518863_

**Comment by eshaan7** [comment]
Will raise PR for this soon.
